### ISSUE TITLE
Mantle Model with nil properties raise RLMException

### ISSUE THREAD

**Issue by andr3a88** [realm/realm-cocoa#2495]
Hi, 
using Realm + Mantle my app crash when the model of mantle has a nil property. For example  twitterUserModel (mantle model) sometimes has **url = nil** and then raise a RLMException:
'No value or default value specified for property 'url' in 'TWUser''. 
The solution, i think is to set the property to the default value. How to?

The method `+ (NSDictionary *)defaultPropertyValues` is called before `initWithMantleModel`, so the property is setted to the default value and after to nil if the property is not present in model.

This is my code:
 
    (id)initWithMantleModel:(MTLTwitterUser *)twitterUserModel {
    self = [super init];
    if(!self) {
        return nil;
    }
    
       //PROBLEM: if a property of twitterUserModel is nil ==> RLMException
    
       self.idStr = twitterUserModel.idStr;
       self.createdAt = twitterUserModel.createdAt;
         .....
         ......
       self.screenName = twitterUserModel.screenName;
       self.utcOffset = twitterUserModel.utcOffset;
       self.verified = twitterUserModel.verified;

      //SOLUTION: set to default value the property setted to nil ???
    
       return self;
    }

**Comment by segiddins** [comment]
In your `initWithMantleModel:` method, you'll want to coalesce any `nil` values to a non-nil default, and you'll have to do so manually because you're explicitly setting the property. Alternatively, you could delegate the Mantle initializer to `initWithValue:`, which uses key-value coding to pull properties off a given object:

```objc
- (instancetype)initWithMantleModel:(MTLTwitterUser *)twitterUserModel {
  return [self initWithValue:twitterUserModel];
}
```

**Comment by andr3a88** [comment]
For now I have resolved with:

    if(twitterUserModel.idStr) 
       self.idStr = twitterUserModel.idStr;

for each property. Not very elegant.

**Comment by jpsim** [comment]
Not much more to say here: our current lack of null support is inelegant for some users, workarounds exist, this will soon be solved in 0.96 with #2458. Closing.
